src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -700,12 +699,16 @@ gl_sarray drawing_classifier::predict(gl_sframe data, std::string output_type) {
     result = result.apply(max_prob_label, class_labels.front().get_type());
 
   } else if (output_type == "probability") {
+
+    if (read_state<flex_int>("num_classes") > 2) {
```

Comment:
Nah, I think this is fundamentally wrong. `num_classes` means the size of probability space. You predict the label with the largest probability in that probability space. It's sitll single class prediction. Correct me if I'm wrong. @nickjong

Reply:
This is an effort to match our behavior in 5.8: https://github.com/apple/turicreate/blob/master/src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py#L941

Reply:
Read the line below. Think twice.

Reply:
Sorry, can you elaborate?

Reply:
https://github.com/apple/turicreate/blob/96a2c5c4035f3177e0d9b2e5d87fd575d816a8a7/src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py#L935

At first glance, `self.classes` is the column name used to retrieve the predicted results.

Reply:
```
 auto max_prob = [=](const flexible_type& ft) {
       const flex_vec& prob_vec = ft.get<flex_vec>();		       const flex_vec& prob_vec = ft.get<flex_vec>();
       auto max_it = std::max_element(prob_vec.begin(), prob_vec.end());		       auto max_it = std::max_element(prob_vec.begin(), prob_vec.end());
       return *max_it;		       return *max_it;
     };		     };

```
does the single class prediction.

Reply:
Would removing this `log_and_throw` be consistent with our behavior in 5.8? How do you recommend we stay consistent with our behavior in 5.8?

Reply:
I don't know what to say already. This will be my last comment on this topic. An even simple and silly question, if we assume this logic was correct, why we wouldn't add the same thing for topk? You haven't figured out the semantic difference of `self.classes` and `num_classes`.

Reply:
per offline discussion, it sounds reasonable to align with the previous behavior. I wonder if we also want to add an output type called `probability_vector`.